core/src/main/scala/kafka/server/ConfigHelper.scala

```diff
@@ -45,8 +45,18 @@ import scala.jdk.OptionConverters.RichOptional
 
 class ConfigHelper(metadataCache: MetadataCache, config: KafkaConfig, configRepository: ConfigRepository) extends Logging {
 
-  def allConfigs(config: AbstractConfig): mutable.Map[String, Any] = {
-    config.originals.asScala.filter(_._2 != null) ++ config.nonInternalValues.asScala
+  def allConfigs(config: AbstractConfig): java.util.Map[String, Object] = {
```

Comment:
Could you please merge `allConfigs` into `createResponseConfig` to remove the temporary collection?